signer-wasm/src/ledger.rs

```diff
@@ -0,0 +1,234 @@
+use filecoin_signer_ledger;
+use js_sys::Promise;
+use serde::{Deserialize, Serialize};
+use wasm_bindgen::prelude::*;
+
+use filecoin_signer_ledger::{APDUTransport, TransportWrapperTrait};
+
+use bip44::BIP44Path;
+
+use crate::utils::{address_to_object, bytes_to_buffer, signature_to_object, Buffer};
+
+// lifted from the `console_log` example
+#[wasm_bindgen]
+extern "C" {
+    #[wasm_bindgen(js_namespace = console)]
+    fn log(s: &str);
+}
+
+/// FilecoinApp Error message
+#[derive(Deserialize, Serialize)]
+pub struct Error {
+    /// Message return code
+    pub return_code: u16,
+    /// Error message
+    pub error_message: String,
+}
+
+#[wasm_bindgen(module = "/transportWrapper.js")]
+extern "C" {
+    pub type TransportWrapper;
+
+    #[wasm_bindgen(method)]
+    pub fn exchange(this: &TransportWrapper, apdu_command: Buffer) -> Promise;
+}
+
+impl TransportWrapperTrait for TransportWrapper {
+    fn exchange_apdu(&self, apdu_command: &[u8]) -> js_sys::Promise {
+        self.exchange(bytes_to_buffer(apdu_command))
+    }
+}
+
+#[wasm_bindgen]
+pub async fn get_version(transport_wrapper: TransportWrapper) -> Promise {
+    let tmp = Box::new(transport_wrapper);
+    let apdu_transport = APDUTransport {
+        transport_wrapper: tmp,
+    };
+
+    // FIXME: handle the error
+    let app = filecoin_signer_ledger::app::FilecoinApp::connect(apdu_transport).unwrap();
+    let v_result = app.get_version().await;
+
+    // FIXME: Do this automatically to simplify this code
+    match v_result {
+        Ok(v) => Promise::resolve(&JsValue::from_serde(&v).unwrap()),
+        Err(err) => {
+            let error = Error {
+                return_code: 0x6f00,
+                error_message: err.to_string(),
+            };
+
+            // FIXME: handle the error
+            Promise::reject(&JsValue::from_serde(&error).unwrap())
+        }
+    }
+}
+
+#[wasm_bindgen]
+pub async fn key_retrieve_from_device(
+    path: String,
+    transport_wrapper: TransportWrapper,
+) -> Promise {
+    let tmp = Box::new(transport_wrapper);
+    let apdu_transport = APDUTransport {
+        transport_wrapper: tmp,
+    };
+
+    // FIXME: handle the error
```

Comment:
same here.. we need an issues to fix error handling, unwrapping will result in crashing when connect fails